odl/solvers/functional/functional.py

```diff
@@ -232,10 +232,10 @@ def bregman(self, point, subgrad=None):
         ----------
         point : element of ``functional.domain``
             Point from which to define the Bregman distance.
-        subgrad : `Operator`, optional
-            A subgradient operator. Operator which that takes an element in
-            `functional.domain` and returns a subdifferential of the functional
-            in that point.
+        subgrad : element of ``functional.domain`` or `Operator`, optional
+            A subgradient of ``functional`` in ``point``, or an operator
+            which that takes an element in ``functional.domain`` and returns a
+            subdifferential of the functional in that point.
```

Comment:
I think we're still not on the same page. What I had in mind was this:
```
subgrad : element of ``functional.domain``, optional
    A subgradient of ``functional`` in the given ``point``.
    The default for ``None`` is ``functional.gradient(point)``,
    which implies that this parameter is mandatory if
    ``functional.gradient`` is not implemented.
```
So no gradient operator anywhere, except for the default subgradient.

Just a nitpicking remark: subgradient = element, subdifferential = set. Some confusion about that here.

Reply:
I'd say that's okay since most of the time it will be called through the `bregman` method, and there it's not outrageous if the creation takes some time.

Reply:
An alternative that I would also find okay would be to make the parameter mandatory and just write in the docs that `functional.gradient(point)` is a valid choice if implemented.

Reply:
I am also in favour of no intelligence and make the `subgrad` mandatory.